REPO-3167: REST API: Produce OpenAPI specification for Rules API

cristianturlica opened the pull request:
- initial version

cristianturlica committed:
REPO-3167: REST API: Produce OpenAPI specification for Rules API
   - initial version

dcaruana commented:
Looking good :)

One minor comment re: the naming of nodeRefId on Rule type. Not sure if this type was already defined, but is it possible to rename nodeRefId to nodeId so we're consistent with the other parts of the API.

Regardless of this issue, this looks good to merge.

cristianturlica committed:
REPO-3167: REST API: Produce OpenAPI specification for Rules API
   - renamed nodeRefId to nodeId so we're consistent with the other parts of the API

cristianturlica commented:
Change done. Thanks for the review.

Branch should remain and not be merged until the implementations are finished.

cristianturlica committed:
REPO-3167: REST API: Produce OpenAPI specification for Rules API
   - renamed nodeId to ruleId for Rule definition

cristianturlica committed:
REPO-3167: REST API: Produce OpenAPI specification for Rules API
   - added rule types endpoint

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Alfresco/rest-api-explorer?pullRequest=52) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Alfresco/rest-api-explorer?pullRequest=52) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Alfresco/rest-api-explorer?pullRequest=52) it.</sub>
